File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/api/graphql/dataloader/PostRepresentationBatchLoader.java
package com.github.arhor.aws.graphql.federation.comments.api.graphql.dataloader;

import com.github.arhor.aws.graphql.federation.comments.generated.graphql.types.Post;
import com.github.arhor.aws.graphql.federation.comments.service.PostRepresentationService;
import com.netflix.graphql.dgs.DgsDataLoader;
import org.springframework.beans.factory.annotation.Qualifier;

import java.util.UUID;
import java.util.concurrent.Executor;

import static org.springframework.boot.autoconfigure.task.TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME;

@DgsDataLoader(maxBatchSize = 50)
public class PostRepresentationBatchLoader extends AbstractMappedBatchLoader<UUID, Post> {

    public PostRepresentationBatchLoader(
        @Qualifier(APPLICATION_TASK_EXECUTOR_BEAN_NAME) Executor executor,
        final PostRepresentationService postService
    ) {
        super(executor, postService::findPostsRepresentationsInBatch);
    }
}

File: app-service-votes/src/main/kotlin/com/github/arhor/aws/graphql/federation/votes/api/graphql/dataloader/PostRepresentationBatchLoader.kt
package com.github.arhor.aws.graphql.federation.votes.api.graphql.dataloader

import com.github.arhor.aws.graphql.federation.votes.generated.graphql.types.Post
import com.github.arhor.aws.graphql.federation.votes.service.PostRepresentationService
import com.netflix.graphql.dgs.DgsDataLoader
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.autoconfigure.task.TaskExecutionAutoConfiguration.APPLICATION_TASK_EXECUTOR_BEAN_NAME
import java.util.UUID
import java.util.concurrent.Executor

@DgsDataLoader(maxBatchSize = 50)
class PostRepresentationBatchLoader(
    @Qualifier(APPLICATION_TASK_EXECUTOR_BEAN_NAME)
    executor: Executor,
    postService: PostRepresentationService,
) : AbstractMappedBatchLoader<UUID, Post>(
    executor = executor,
    loaderFn = postService::findPostsRepresentationsInBatch,
)
